src/VisionToken.new.sol

```diff
@@ -0,0 +1,206 @@
+    bytes32 internal constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    /// @notice Role for upgrading the contract.
+    bytes32 public constant UPGRADER_ROLE = keccak256("UPGRADER_ROLE");
+
+    /**
+     * @dev Emitted when `amount` tokens are minted by `MINTER_ROLE` and assigned to `to`.
+     */
+    event Mint(address indexed minter, address indexed to, uint256 amount);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    // solhint-disable-next-line func-visibility
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @dev criticalOps receives all existing tokens and is the owner of underlying VisionBaseToken
+     */
+    function initialize(
+        uint256 initialSupply,
+        address defaultAdmin,
+        address criticalOps,
+        address minter,
+        address pauser,
+        address upgrader
+    ) public initializer {
+        // __VisionBaseToken_init() also initializes Upgradeable variants of ERC165, Ownable, ERC20, ERC20Permit 
+        __VisionBaseToken_init(_NAME, _SYMBOL, _DECIMALS, criticalOps);
+        __ERC20Pausable_init();
+        __AccessControl_init();
+        __UUPSUpgradeable_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, defaultAdmin);
+        _grantRole(CRITICAL_OPS, criticalOps);
+        _grantRole(MINTER_ROLE, minter);
+        _grantRole(PAUSER_ROLE, pauser);
+        _grantRole(UPGRADER_ROLE, upgrader);
+
+        ERC20Upgradeable._mint(super.getOwner(), initialSupply);
+        // Contract is paused until forwarder is set
+        _pause();
+    }
+
+    /**
+     * @notice Mints tokens to a specified address.
+     * @dev Only callable by accounts with the `MINTER_ROLE`.
+     * Emits a {Mint} event with `minter` set to the address that initiated the minting,
+     * `to` set to the recipient's address, and `amount` set to the amount of tokens minted.
+     * Reverts with "VisionToken: Mint amount is zero" if `amount` is zero.
+     * Requirement: the contract must not be paused. {ERC20PausableUpgradeable-_update} enforces it.
+     * @param to The address to receive the minted tokens.
+     * @param amount The amount of tokens to mint.
+     */
+    function mint(address to, uint256 amount) external onlyRole(MINTER_ROLE) {
+        require(amount != 0, "VisionToken: Mint amount is zero");
+        _mint(to, amount);
+        emit Mint(_msgSender(), to, amount);
+    }
+
+    /**
+     * @dev See {VisionBaseToken-onlyVisionForwarder}
+     */
+    modifier onlyVisionForwarder() override {
+        require(
+            msg.sender == getVisionForwarder(),
+            "VisionToken: caller is not the VisionForwarder"
+        );
+        _;
+    }
+
+    /**
+     * @dev See {Pausable-_pause)
+     */
+    function pause() external onlyRole(PAUSER_ROLE) {
+        _pause();
+    }
+
+    /**
+     * @dev See {Pausable-_unpause)
+     */
+    function unpause() external onlyRole(CRITICAL_OPS) {
+        require(
```

Comment:
@jacekv  Should we allow unpause without forwarder being set ? to decouple dependency of available forwarder ?

Reply:
done as agreed

Reply:
additionally I have removed pausing of the token on deployment(initialization), as we are not coupled with availability of forwarder.